src/AdyenCheckoutContext.js

```diff
@@ -1,24 +1,79 @@
+import {
+  AdyenDropIn,
+  // @ts-ignore
+  AdyenComponent,
+  getNativeComponent,
+  // @ts-ignore
+  PaymentMethodsResponse,
+} from './AdyenNativeModules';
 import { NativeEventEmitter } from 'react-native';
 
 const AdyenCheckoutContext = createContext({
   start: (/** @type {string} */ typeName) => {},
   config: {},
-  paymentMethods: /** @type {PaymentMethodsResponse} */ undefined,
+  paymentMethods: /** @type {PaymentMethodsResponse=} */ undefined,
 });
 
-const AdyenCheckout = ({
-  config,
-  paymentMethods,
-  onSubmit,
-  onComplete,
-  onFail,
-  onProvide,
-  children,
-}) => {
+/**
+ *
+ * @callback OnSubmitFunction
+ * @param {*}  data Payment details collected by component. Your server should use it to make the Adyen API `/payments` request.
+ * @param {AdyenComponent & AdyenDropIn}  nativeComponent Native component that performing payment.
+ * @returns {void}
+ */
+
+/**
+ *
+ * @callback OnAdditionalDetailsFunction
+ * @param {*} data Additional payment challenge details.
+ * @param {AdyenComponent}  nativeComponent Native component that performing payment.
+ * @returns {void}
+ */
+
+/**
+ * @callback OnCompletedFunction
+ * @param {AdyenComponent}  nativeComponent Native component that performing payment.
+ * @returns {void}
+ */
+
+/**
+ *
+ * @callback OnErrorFunction
+ * @param {{ message: string; errorCode: string; }} error Reason for payment termination.
+ * @param {AdyenComponent}  nativeComponent Native component that performing payment.
+ * @returns {void}
+ */
+
+/**
+ * @typedef {Object} AdyenCheckoutProp
+ * @property {*} config Collection of all necessary configurations
+ * @property {PaymentMethodsResponse } paymentMethods JSON response from Adyen API `\paymentMethods`
+ * @property {OnSubmitFunction} onSubmit Event callback, called when the shopper selects the Pay button and payment details are valid.
+ * @property {OnAdditionalDetailsFunction=} onAdditionalDetails Event callback, called when a payment method requires more details, for example for native 3D Secure 2, or native QR code payment methods.
+ * @property {OnCompletedFunction=} onComplete Event callback, called when a shopper finishes the flow (Voucher payments only).
+ * @property {OnErrorFunction=} onError Event callback, called when payment about to be terminate.
+ * 
+ * @property {OnAdditionalDetailsFunction=} onProvide Deprecated, use 'onAdditionalDetails' instead.
+ * @property {OnErrorFunction=} onFail Deprecated, use 'onError' instead.
+
+ * @property {React.ReactNode} children Inline elements.
+ */
+
+/** @param {AdyenCheckoutProp} props */
+const AdyenCheckout = (props) => {
+  /** @type {import('react').MutableRefObject<import('react-native').EventSubscription[]>} */
   const subscriptions = useRef([]);
+  const { config, paymentMethods, onSubmit, onComplete, children } = props;
+
+  let onError = props.onError == undefined ? props.onFail : props.onError;
+  let onAdditionalDetails =
```

Comment:
Always use `const` instead of `let`. If the compiler complain, then you fix it. Most (if not all) the cases you are going to use `const`

Reply:
Fixed in [30f2220](https://github.com/Adyen/adyen-react-native/pull/155/commits/30f2220b7d503574eb14366aee50dcc192488eae)